phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -3546,9 +3586,6 @@ public Void call() throws Exception {
                                 success = true;
                                 return null;
                             }
-                            nSequenceSaltBuckets = ConnectionQueryServicesImpl.this.props.getInt(
```

Comment:
Moving initialization of nSequenceSaltBuckets to the constructor since SKIP_SYSTEM_TABLES_EXISTENCE_CHECK can be set to true and we return early at L3587 and this will leave nSequenceSaltBuckets initiliazed to default value i.e 0.